src/kvtree.c

```diff
@@ -1511,7 +1574,7 @@ int kvtree_lock_open_read(const char* file, int* fd, kvtree* hash)
 
   /* open file with lock for read / write access */
   mode_t mode_file = kvtree_getmode(1, 1, 0);
-  *fd = kvtree_open_with_lock(file, O_RDWR | O_CREAT, mode_file);
+  *fd = kvtree_open_with_lock(file, O_RDWR | O_CREAT, mode_file, 0);
```

Comment:
We need to make this a write lock.  This function is called by the user as the front part of a read-modify-write operation.

Reply:
Ok, I'll change it back and will update the documentation for `kvtree_lock_open_read()` to describe why it's getting a write lock.

Reply:
Yeah, thanks.  This function is the open half of this pair, which the caller uses together:
```
/** given a filename and hash, lock the file, open it, and read it into hash, set fd to the opened file descriptor */
int kvtree_lock_open_read(const char* file, int* fd, kvtree* hash);

/** given a filename, an opened file descriptor, and a hash, overwrite file with hash, close, and unlock file */
int kvtree_write_close_unlock(const char* file, int* fd, const kvtree* hash);
```